### PR TITLE
Set the envoy retry policy for external-authz filter

### PR DESCRIPTION
**Pull request by Warashi**
**What this PR does / why we need it**:

I saw the envoy access log sometimes records UAEX (UnauthorizedExternalService) errors even if the ext-authz service returns them as OK.
I checked the admin port of the envoy and realized the tx_reset had occurred.
I don't know how to reduce the tx_reset, so I want to try the retry_policy.
When we use the retry_policy, we have to bump the envoy version.
So, I bump the envoy version to the latest, then I configured the retry_policy.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

- **How are users affected by this change**:
- **Is this breaking change**:
- **How to migrate (if breaking change)**:

### CODE REVIEW

File: manifests/pipecd/templates/envoy-configmap.yaml [hunk 1]
```diff
@@ -38,18 +38,31 @@ data:
...
+                  "@type": type.googleapis.com/envoy.extensions.filters.http.cors.v3.Cors
 {{- end }}
               - name: envoy.filters.http.grpc_stats
                 typed_config:
                   "@type": type.googleapis.com/envoy.extensions.filters.http.grpc_stats.v3.FilterConfig
                   stats_for_all_methods: true
                   enable_upstream_stats: true
               - name: envoy.filters.http.router
+                typed_config:
+                  "@type": type.googleapis.com/envoy.extensions.filters.http.router.v3.Router
```
Comment (Warashi) [inline manifests/pipecd/templates/envoy-configmap.yaml:65]: We have to set these to bump the envoy version

File: manifests/pipecd/templates/deployment.yaml [hunk 2]
```diff
@@ -25,7 +25,7 @@ spec:
       {{- end }}
       containers:
         - name: envoy
-          image: envoyproxy/envoy-alpine:{{ .Values.gateway.imageTag }}
+          image: envoyproxy/envoy:{{ .Values.gateway.imageTag }}
```
Comment (Warashi) [inline manifests/pipecd/templates/deployment.yaml:28]: envoy-alpine has stopped updating 2 years ago

**Review by khanhtc1202 — APPROVED** [review]
Thanks for detailed explanation 🫡

**Review by ffjlabo — APPROVED** [review]
LGTM 🙆‍♂️

**Pull request merged**